api/backend/blueflood/blueflood.go

```diff
@@ -115,17 +115,71 @@ func (b *Blueflood) FetchSeries(metric api.TaggedMetric, predicate api.Predicate
 	}, nil
 }
 
+func addMetricPoint(metricPoint MetricPoint, field func(MetricPoint) float64, timerange api.Timerange, buckets [][]float64) error {
+	value := field(metricPoint)
+	timestamp := metricPoint.Timestamp
+	// The index to assign within the array is computed using the timestamp.
+	// It rounds to the nearest index.
+	index := (timestamp - timerange.Start() + timerange.Resolution()/2) / timerange.Resolution()
+	if index < 0 || index >= int64(timerange.Slots()) {
+		return errors.New("index out of range")
+	}
+	buckets[index] = append(buckets[index], value)
+	return nil
+}
+
+func bucketsFromMetricPoints(metricPoints []MetricPoint, resultField func(MetricPoint) float64, timerange api.Timerange) [][]float64 {
+	buckets := make([][]float64, timerange.Slots())
+	// Make the buckets:
+	for i := range buckets {
+		buckets[i] = []float64{}
+	}
+	for _, point := range metricPoints {
+		addMetricPoint(point, resultField, timerange, buckets)
+	}
+	return buckets
+}
+
 func (b *Blueflood) fetchSingleSeries(metric api.GraphiteMetric, sampleMethod api.SampleMethod, timerange api.Timerange) ([]float64, error) {
 	// Use this lowercase of this as the select query param. Use the actual value
 	// to reflect into result MetricPoints to fetch the correct field.
-	var selectResultField string
+	var (
```

Comment:
store this in `map[sampleMethod]HelperStruct`, where `HelperStruct` has the previous 3 fields?